diff -Nur wvstreams-4.0.2/crypto/wvtripledes.cc wvstreams-4.0.2.new_ssl/crypto/wvtripledes.cc
--- wvstreams-4.0.2/crypto/wvtripledes.cc	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.new_ssl/crypto/wvtripledes.cc	2006-03-05 14:36:11.000000000 +0100
@@ -103,7 +103,7 @@
 	while (len >= 8)
 	{
 #if OPENSSL_VERSION_NUMBER >= 0x0090705FL
-	    DES_ecb3_encrypt(data, crypt,
+	    DES_ecb3_encrypt((unsigned char (*)[8])data, (unsigned char (*)[8])crypt,
 			     &deskey1, &deskey2, &deskey3,
 			     mode == ECBEncrypt ? DES_ENCRYPT : DES_DECRYPT);
 #else
diff -Nur wvstreams-4.0.2/crypto/wvx509.cc wvstreams-4.0.2.new_ssl/crypto/wvx509.cc
--- wvstreams-4.0.2/crypto/wvx509.cc	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.new_ssl/crypto/wvx509.cc	2006-03-05 14:35:18.000000000 +0100
@@ -1163,11 +1163,11 @@
 		{
 		    void *ext_data = NULL;
 		    if (method->it) 
-			ext_data = ASN1_item_d2i(NULL, &ext->value->data, 
+			ext_data = ASN1_item_d2i(NULL, (const unsigned char**)&ext->value->data, 
 						ext->value->length, 
 						ASN1_ITEM_ptr(method->it));
 		    else
-			ext_data = method->d2i(NULL, &ext->value->data, 
+			ext_data = method->d2i(NULL, (const unsigned char**)&ext->value->data, 
 					      ext->value->length);
 		    
 		    if (method->i2s)
