--- wvstreams-4.0.2/include/unilistgen.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/include/unilistgen.h	2006-01-18 21:22:04.000000000 +0100
@@ -31,7 +31,7 @@
 {
 public:
     UniListGen(UniConfGenList *_l);
-    virtual UniListGen::~UniListGen() { delete l; }
+    virtual ~UniListGen() { delete l; }
 
     UniConfGenList *l;
     UniConfGenList::Iter i;
--- wvstreams-4.0.2/include/wvhttppool.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/include/wvhttppool.h	2006-01-18 21:22:20.000000000 +0100
@@ -57,7 +57,7 @@
     WvString linkname;
     WvUrl url;
 
-    WvUrlLink::WvUrlLink(WvStringParm _linkname, WvStringParm _url)
+    WvUrlLink(WvStringParm _linkname, WvStringParm _url)
 	: linkname(_linkname), url(_url)
     {}
 };
--- wvstreams-4.0.2/xplc/include/xplc/IObject.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/xplc/include/xplc/IObject.h	2006-07-27 10:56:07.470497750 +0200
@@ -61,6 +61,8 @@
 class IObject {
   UNSTABLE_INTERFACE
 public:
+  virtual ~IObject() { }
+
   /**
    * Indicate you are using this object.  This increases the reference
    * count of the object by one.  Usually, when the reference count
--- wvstreams-4.0.2/crypto/wvtripledes.cc	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.gcc4/crypto/wvtripledes.cc	2006-07-27 20:08:31.000000000 +0000
@@ -102,7 +102,7 @@
 	// ECB works 64bits at a time
 	while (len >= 8)
 	{
-#if OPENSSL_VERSION_NUMBER >= 0x0090705FL
+#if (OPENSSL_VERSION_NUMBER >= 0x0090705FL) && (OPENSSL_VERSION_NUMBER < 0x0090708FL)
 	    DES_ecb3_encrypt(data, crypt,
 			     &deskey1, &deskey2, &deskey3,
 			     mode == ECBEncrypt ? DES_ENCRYPT : DES_DECRYPT);
--- wvstreams-4.0.2/crypto/wvx509.cc	2006-07-28 12:50:53.000000000 +0000
+++ wvstreams-4.0.2.gcc4/crypto/wvx509.cc	2006-07-28 12:50:14.000000000 +0000
@@ -1162,6 +1162,17 @@
 		else
 		{
 		    void *ext_data = NULL;
+#if OPENSSL_VERSION_NUMBER >= 0x0090708FL
+		    if (method->it) 
+			ext_data = ASN1_item_d2i(NULL, 
+				(const unsigned char**) &ext->value->data, 
+						ext->value->length, 
+						ASN1_ITEM_ptr(method->it));
+		    else
+			ext_data = method->d2i(NULL,
+				(const unsigned char**) &ext->value->data, 
+					      ext->value->length);
+#else
 		    if (method->it) 
 			ext_data = ASN1_item_d2i(NULL, &ext->value->data, 
 						ext->value->length, 
@@ -1169,6 +1180,7 @@
 		    else
 			ext_data = method->d2i(NULL, &ext->value->data, 
 					      ext->value->length);
+#endif
 		    
 		    if (method->i2s)
 		    {
