--- wvstreams-4.0.2/include/unilistgen.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/include/unilistgen.h	2006-01-18 21:22:04.000000000 +0100
@@ -31,7 +31,7 @@
 {
 public:
     UniListGen(UniConfGenList *_l);
-    virtual UniListGen::~UniListGen() { delete l; }
+    virtual ~UniListGen() { delete l; }
 
     UniConfGenList *l;
     UniConfGenList::Iter i;
--- wvstreams-4.0.2/include/wvhttppool.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/include/wvhttppool.h	2006-01-18 21:22:20.000000000 +0100
@@ -57,7 +57,7 @@
     WvString linkname;
     WvUrl url;
 
-    WvUrlLink::WvUrlLink(WvStringParm _linkname, WvStringParm _url)
+    WvUrlLink(WvStringParm _linkname, WvStringParm _url)
 	: linkname(_linkname), url(_url)
     {}
 };
--- wvstreams-4.0.2/xplc/include/xplc/IObject.h	2005-03-24 03:01:23.000000000 +0100
+++ wvstreams-4.0.2.gcc4/xplc/include/xplc/IObject.h	2006-07-27 10:56:07.470497750 +0200
@@ -61,6 +61,8 @@
 class IObject {
   UNSTABLE_INTERFACE
 public:
+  virtual ~IObject() { }
+
   /**
    * Indicate you are using this object.  This increases the reference
    * count of the object by one.  Usually, when the reference count
