--- wvstreams-4.0.2/vars.mk~	2006-07-30 23:51:49.000000000 +0000
+++ wvstreams-4.0.2/vars.mk	2006-07-31 00:11:36.000000000 +0000
@@ -139,15 +139,15 @@
 libwvbase.so: LIBS+=-lxplc-cxx
 
 ifneq ("$(with_fam)", "no")
-  libwvstreams.so: -lfam
+  libwvstreams.so: LIBS+=-lfam
 endif
 
 ifneq ("$(with_openslp)", "no")
-  libwvstreams.so: -lslp
+  libwvstreams.so: LIBS+=-lslp
 endif
 
 ifneq ("$(with_pam)", "no")
-  libwvstreams.so: -lpam
+  libwvstreams.so: LIBS+=-lpam
 endif
 
 LDLIBS := -lgcc $(LDLIBS) \
diff -Nur wvstreams-4.0.2/xplc/config/config.mk.in wvstreams-4.0.2.mk/xplc/config/config.mk.in
--- wvstreams-4.0.2/xplc/config/config.mk.in	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.mk/xplc/config/config.mk.in	2006-07-31 00:53:17.000000000 +0000
@@ -42,4 +42,4 @@
 LN_S:=@LN_S@
 LIBS:=@LIBS@
 CVS2CL:=@CVS2CL@
-
+LDFLAGS:=@LDFLAGS@
diff -Nur wvstreams-4.0.2/xplc/config/rules.mk wvstreams-4.0.2.mk/xplc/config/rules.mk
--- wvstreams-4.0.2/xplc/config/rules.mk	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.mk/xplc/config/rules.mk	2006-07-31 00:54:03.000000000 +0000
@@ -40,7 +40,7 @@
 	$(RANLIB) $@
 
 %.so:
-	$(LINK.cc) $(SHARED) $^ -o $@
+	$(LINK.cc) $(SHARED) $^ -o $@ $(LIBS)
 
 %.dll:
 	$(LINK.cc) $(SHARED) $^ -o $@
