diff -Nur wvstreams-4.0.2/vars.mk wvstreams-4.0.2.cflags/vars.mk
--- wvstreams-4.0.2/vars.mk	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.cflags/vars.mk	2006-02-26 23:56:16.000000000 +0000
@@ -103,12 +103,6 @@
 #CFLAGS+=-Werror
 endif
 
-ifneq ("$(enable_optimization)", "no")
-CXXFLAGS+=-O2
-#CXXFLAGS+=-felide-constructors
-CFLAGS+=-O2
-endif
-
 ifneq ("$(enable_warnings)", "no")
 CXXFLAGS+=-Wall -Woverloaded-virtual
 CFLAGS+=-Wall
diff -Nur wvstreams-4.0.2/wvrules.mk wvstreams-4.0.2.cflags/wvrules.mk
--- wvstreams-4.0.2/wvrules.mk	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.cflags/wvrules.mk	2006-02-26 23:55:55.000000000 +0000
@@ -78,12 +78,6 @@
   #CFLAGS+=-Werror
 endif
 
-ifneq ("$(enable_optimization)", "no")
-  CXXFLAGS+=-O2
-  #CXXFLAGS+=-felide-constructors
-  CFLAGS+=-O2
-endif
-
 ifneq ("$(enable_warnings)", "no")
   CXXFLAGS+=-Wall -Woverloaded-virtual
   CFLAGS+=-Wall
diff -Nur wvstreams-4.0.2/xplc/configure.ac wvstreams-4.0.2.cflags/xplc/configure.ac
--- wvstreams-4.0.2/xplc/configure.ac	2005-03-24 02:01:23.000000000 +0000
+++ wvstreams-4.0.2.cflags/xplc/configure.ac	2006-02-26 23:56:55.000000000 +0000
@@ -114,10 +114,6 @@
     AC_DEFINE(ENABLE_LOADER, 1, [Define if you want to enable dynamic loading.])
 fi
 
-if test "$enable_optimization" != "no"; then
-    CXXFLAGS="$CXXFLAGS -O2"
-fi
-
 if test -z "$enable_debug"; then
     enable_debug=no
 fi
